test/index.js

```diff
@@ -451,4 +473,79 @@ describe('KeyringController', function () {
       );
     });
   });
+
+  describe('Manifest v3 changes', function () {
+    it('can still verify password', async function () {
+      // Log the user out
+      await keyringController.setLocked();
+
+      // Attempt to verify the password
+      const result = await keyringController.verifyPassword(password);
+      expect(result).toBeTruthy();
+    });
+
+    it('can still unlock with password after being migrated and locked', async function () {
+      // Log the user out
+      await keyringController.setLocked();
+
+      // Attempt to unlock the keychaing via old password
+      const result = await keyringController.submitPassword(password);
+      expect(result).toBeTruthy();
+
+      // Ensure the new vault value has separator and salt
+      const { vault, salt } = keyringController.store.getState();
+      expect(vault).toContain(':::');
+      expect(salt).toBeTruthy();
+    });
+
+    // TODO:  THIS SHOULD BE FAILING
+    it('does not unlock with incorrect encryption key', async function () {
+      await keyringController.submitPassword(password);
+
+      // Log the user out
+      await keyringController.setLocked();
+
+      // Attempt to login with an incorrect encryption key
+      const encryptionKey = await keyringController.submitEncryptedKey(
+        'FAKE KEY',
+      );
+
+      expect(encryptionKey).rejects.toThrow();
+    });
+
+    it('unlocks with correct encryption key', async function () {
+      await keyringController.submitPassword(password);
+
+      const encryptionKey = keyringController.getEncryptedKey();
+
+      // Log the user out
+      await keyringController.setLocked();
+
+      // Attempt to login with an incorrect encryption key
+      const key = await keyringController.submitEncryptedKey(encryptionKey);
+
+      expect(key).toBeTruthy();
+    });
+
+    it('removes encryption key when locked', async function () {
+      await keyringController.setLocked();
+
+      expect(keyringController.getEncryptedKey()).toBeUndefined();
+    });
+
+    it('rotates encrypted keys between logins', async function () {
+      // Log in with correct password, which will trigger generation of key
+      await keyringController.submitPassword(password);
+
+      const currentKey = keyringController.getEncryptedKey();
+
+      await keyringController.setLocked();
+
+      await keyringController.submitPassword(password);
+
+      const newKey = keyringController.getEncryptedKey();
+
+      expect(currentKey === newKey).toBe(false);
```

Comment:
if you use a "not equal" assertion here, you'll get both values in test failure output when it fails, instead of "expected true to be false"

Reply:
Good improvement!